### DISCUSSION TITLE
Simple Script-Based CloudStack Installer

### DISCUSSION THREAD

**Discussion by yadvr** [apache/cloudstack#9953] [Ideas]
Test and explore https://github.com/rohityadavcloud/c8k.in, add/report issues & send pull requests.

The ultimate goal is to allow (in future) anybody wanting to try CloudStack by simply doing:

```
curl -sL https://c8k.in/stall.sh | bash
```

Update: moved here https://github.com/apache/cloudstack-installer

**Comment by baltazorbest** [comment]
Add support for VLAN and bonding. Currently, the script removes the existing VLAN and bonding configuration from the 01-netcfg.yaml file. Thanks.

**Reply by soulfulgent** [reply]
I would like to suuggest the script get rid of mysql and go with **Mariadb** and use the **"mariadb-secure-installation"** script for for locking down.  Mariadb was born in 2009 as a fork of mysql. I'm not a developer, but have some scripting knowledge I've picked up in the last couple of decades working in IT. One thing I'm aware of is that **MariaDB is a lot faster than mysql, uses less resources...etc.**

**Secondly, will the script work with Ubuntu 25.10.x.** UltimetEly, I would like to see the cloudstack script install on Ubuntu 25.x using MariaDB with "mariadb-secure-installation" script 


**Reasons why to use MariaDB; see links below;**

**Why Maria Left MySQL: The Database Drama Nobody Talks About**
https://medium.com/@mehdibafdil/why-maria-left-mysql-b703241d72ac

**benchmarks;**
https://willthefarmer.github.io/index.html#Maria-Benchmark

**Fastest Open-Source Databases**
https://datasystemreviews.com/fastest-open-source-databases.html

Kind Regards,
Mike

**Comment by taocnrut** [comment]
don't hardcode ens0 interface name :)

**Comment by m-dhellin** [comment]
@rohityadavcloud I am currently testing it on WSL Ubuntu 22.04.5 LTS

**Comment by patrix** [comment]
Testing on Ubuntu 24.04

**Comment by NuxRo** [comment]
```  gateway=$(ip route show 0.0.0.0/0 dev ens3 | cut -d\  -f3)```
replace ens3 with $interface

**Reply by patrix** [reply]
or...
`gateway=$(ip route show | grep "default via" | cut -d' ' -f3)`

**Reply by m-dhellin** [reply]
I have some struggle with the gateway. I will try your suggestions, thanks.

**Comment by patrix** [comment]
```shell
generate_password() {
    openssl </dev/null >/dev/null 2>/dev/null
    if [ $? -eq 0 ]
    then
        passwd=`openssl rand -base64 40 | tr -d '/' | cut -c1-32  `
    fi
    if [ -z "$passwd" ]
    then
        passwd=`dd if=/dev/urandom bs=32 count=1 2>/dev/null | od -x | head -n 1 | tr -d ' ' | cut -c8-40`
    fi
    if [ -z "$passwd" ]
    then
        passwd="password"
    fi
    echo "$passwd"
}
PASSWD=$(generate_password) # Random password with 32 characters
```

**Comment by weizhouapache** [comment]
may support more OSes , for example RHEL 8/9 and variants ?

**Comment by m-dhellin** [comment]
On WSL, added this line just after the creation of /etc/netplan/01-netcfg.yaml to fix a permission issue :
`chmod go-rwx /etc/netplan/01-netcfg.yaml`

**Comment by m-dhellin** [comment]
The installer keeps breaking the Internet connection of WSL, I’m still trying to figure out why.

**Comment by bvitnik** [comment]
I have made a modest pull request that improves some network handling (including hardcoded ens3) and some mysql handling. This was (unfortunately) tested on VMware hosted VM with nested virtualization :)

**Reply by yadvr** [reply]
Thanks very much @bvitnik merged your PR https://github.com/rohityadavcloud/c8k.in/pull/1 !

**Comment by yadvr** [comment]
This is now moved to https://github.com/apache/cloudstack-installer
